src/dataset/infrastructure/repositories/DatasetJSDataverseRepository.ts

```
@@ -364,4 +365,9 @@ export class DatasetJSDataverseRepository implements DatasetRepository {
       .execute(datasetId, includeMDC)
       .then((jsDatasetDownloadCount) => jsDatasetDownloadCount)
   }
+  deleteDatasetDraft(datasetId: string | number): Promise<void> {
+    return deleteDatasetDraft.execute(datasetId).catch((error: WriteError) => {
+      throw new Error(error.message)
```

could you please throw the error as it is? otherwise you are creating an `Error` and losing the 
 original `WriteError` instance